### PR TITLE
Pages critical security fix

### PR DESCRIPTION
**Pull request by mpuhacz**
**Reference person:** @mpuhacz
**Description:**

Decorators checking that current user is authenticated and logged in as administrator.

### CODE REVIEW

**Comment by magul** [comment]
@mpuhacz I've just issued #561 to solve Travis failing on build.

I have done a lot with Your pages view, so You're pull request will need to be rebased.

In meanwhile I will do proper review on You're code.

And BTW if Travis will be solved (and it is going to be) we will only accept PR that are not causing Travis to fail.

File: apps/volontulo/views/pages.py [hunk 1]
```diff
@@ -12,19 +14,24 @@
 from apps.volontulo.models import Page
 
 
-# pylint: disable=too-many-ancestors;missing-docstring
+# pylint: disable=too-many-ancestors,missing-docstring
 class PageList(ListView):
     model = Page
     template_name = 'pages/page_list.html'
 
+    @method_decorator(user_passes_test(
```
Comment (magul) [inline apps/volontulo/views/pages.py]: You have the same `lambda` in 4 places. Please create named function somewhere and pass it here.

Second thing here is that pylint is failing on Your 8 spaces indentation.

File: apps/volontulo/views/pages.py [hunk 2]
```diff
@@ -42,8 +49,13 @@ def form_valid(self, form):
         self.object.save()
         return redirect(self.get_success_url())
 
+    @method_decorator(user_passes_test(
```
Comment (magul) [inline apps/volontulo/views/pages.py]: see above.

File: apps/volontulo/views/pages.py [hunk 3]
```diff
@@ -54,8 +66,13 @@ class PageEdit(UpdateView):
     template_name = 'pages/page_edit_form.html'
     success_url = reverse_lazy('pages_list')
 
+    @method_decorator(user_passes_test(
```
Comment (magul) [inline apps/volontulo/views/pages.py]: see above.

File: apps/volontulo/views/pages.py [hunk 4]
```diff
@@ -66,3 +83,8 @@ def get(self, request, *args, **kwargs):
         without confirmation
         """
         return self.post(request, *args, **kwargs)
+
+    @method_decorator(user_passes_test(
```
Comment (magul) [inline apps/volontulo/views/pages.py]: ditto

File: apps/volontulo/views/pages.py [hunk 5]
```diff
@@ -12,19 +14,24 @@
...
 
-# pylint: disable=too-many-ancestors;missing-docstring
+# pylint: disable=too-many-ancestors,missing-docstring
 class PageList(ListView):
     model = Page
     template_name = 'pages/page_list.html'
 
+    @method_decorator(user_passes_test(
+            lambda u: u.is_authenticated() and u.userprofile.is_administrator))
+    def dispatch(self, *args, **kwargs):
```
Comment (magul) [inline apps/volontulo/views/pages.py:27]: Maybe it's a little bit too strict but our pylint configuration right now requires docstrings for every method.

**Comment by mpuhacz** [comment]
@magul thanks for the review!

**Pull request merged**